### ISSUE TITLE
Investigate why `grandpa`  is sending the most message out of all protocols

### ISSUE THREAD

**Issue by alexggh** [paritytech/polkadot-sdk#5415]
Looking at the sending rate per second on kusama nodes, grandpa seems to be sending 3 times more messages than any of the other network protocols, see graph.

<img width="1255" alt="Screenshot 2024-08-20 at 14 30 53" src="https://github.com/user-attachments/assets/0773367c-3f6d-4ef1-b12d-52e895141b09">

I would like to understand if this is just an outcome of how the protocols works or something else.

**Comment by sandreim** [comment]
One thing I remember was that it doesn't use a topology, so it sends gossip to all other validators. This could explain it, but TBH I don't have much insight into how GRANDPA is implemented.  

cc @andresilva @bkchr

**Comment by bkchr** [comment]
Yeah grandpa is not using any smart gossip. There is a lot of room for improvement.

**Comment by burdges** [comment]
We could afaik send grandpa messages using the same topology as approval assignments & votes.  We could improve that topology imho, but that's orthogonal.

We'd several grandpa rounds per tranche too, which lowers latency slightly, but likely achieves little.

**Comment by acatangiu** [comment]
Related issue with some ideas explored: https://github.com/paritytech/polkadot-sdk/issues/3896
